### PR TITLE
Drawer bug typeerror on resize

### LINKED ISSUE #7421: Bug - Drawer - Uncaught TypeError on resize
**Describe the problem**
Error appears in the console when resizing a Drawer

**How do you reproduce the problem?**
git clone git@github.com:KevinFCormier/ui-components.git
cd ui-components
npm ci && npm start

1. Navigate to http://localhost:9000/?path=/story/drawer--drawer&args=isResizable:true in browser
1. Click "Open drawer" button
1. Open developer tools
1. Resize the drawer

Observe in console: Uncaught TypeError: Cannot read properties of undefined (reading 'current')

drawerContentRef is undefined in DrawerPanelContent

**Expected behavior**
No console errors.

**Is this issue blocking you?**
No.

**Screenshots**
image

**What is your environment?**
 - OS: macOS
 - Browser: Chrome 101.0.4951.54, Firefox 100.0

**What is your product and what release date are you targeting?**
Red Hat Advanced Cluster Management; not yet determined

**Any other information?**
This problem started with @patternfly/react-core 4.202

### PR DESCRIPTION
**Pull request by wise-king-sullyman**
**What**: Closes #7421

**Additional issues**: N/A

The root cause of the bug here is that `DrawerPanelContent` currently expects to be a direct child of `DrawerMain` and receive `drawerContentRef` from `DrawerContent`, but if the panel content is wrapped in a context that prop gets added to the context rather than `DrawerPanelContent`.

This resolution adds `DrawerContentRef` to the context for the drawer so that its value can be accessed by `DrawerPanelContent` regardless of what it may be wrapped in.

<details>
<summary>Easy way to test:</summary>

1. Open the [drawer examples on the live site](https://www.patternfly.org/v4/components/drawer/) and the [drawer examples on the preview for this branch](https://patternfly-react-pr-7531.surge.sh/components/drawer/) in separate tabs
2. Paste the code below into the code editor for one of the examples in each tab (it wraps `DrawerPanelContent` in a context)

```js
import React from 'react';
import {
  Drawer,
  DrawerPanelContent,
  DrawerContent,
  DrawerContentBody,
  DrawerHead,
  DrawerActions,
  DrawerCloseButton,
  Button
} from '@patternfly/react-core';

export const DrawerResizableOnRight: React.FunctionComponent = () => {
  const [isExpanded, setIsExpanded] = React.useState(false);
  const drawerRef = React.useRef<HTMLDivElement>();

  const TestContext = React.createContext({});

  const onExpand = () => {
    drawerRef.current && drawerRef.current.focus();
  };

  const onClick = () => {
    setIsExpanded(!isExpanded);
  };

  const onCloseClick = () => {
    setIsExpanded(false);
  };

  const onResize = (newWidth: number, id: string) => {
    // eslint-disable-next-line no-console
    console.log(`${id} has new width of: ${newWidth}`);
  };

  const panelContent = (
    <TestContext.Provider value={{}}>
      <DrawerPanelContent
        isResizable
        onResize={onResize}
        id="right-resize-panel"
        defaultSize={'500px'}
        minSize={'150px'}
      >
        <DrawerHead>
          <span tabIndex={isExpanded ? 0 : -1} ref={drawerRef}>
            drawer-panel
          </span>
          <DrawerActions>
            <DrawerCloseButton onClick={onCloseClick} />
          </DrawerActions>
        </DrawerHead>
      </DrawerPanelContent>
    </TestContext.Provider>
  );

  const drawerContent = 'Lorem ipsum dolor sit amet';

  return (
    <React.Fragment>
      <Button aria-expanded={isExpanded} onClick={onClick}>
        Toggle drawer
      </Button>
      <Drawer isExpanded={isExpanded} onExpand={onExpand} position="right">
        <DrawerContent panelContent={panelContent}>
          <DrawerContentBody>{drawerContent}</DrawerContentBody>
        </DrawerContent>
      </Drawer>
    </React.Fragment>
  );
};
```
3. Open the console in your dev tools
4. Toggle the edited drawer example in each tab and attempt to resize it
5. View the errors that result in the live drawer example and verify that they don't occur in the preview branch example
</details>

### CODE REVIEW

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx [hunk 1]
```diff
@@ -37,8 +37,6 @@ export interface DrawerPanelContentProps extends React.HTMLProps<HTMLDivElement>
   };
   /** Color variant of the background of the drawer panel */
   colorVariant?: DrawerColorVariant | 'light-200' | 'default';
-  /** @hide Internal ref to drawer content */
-  drawerContentRef?: React.RefObject<HTMLDivElement>;
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:41]: One question I have, can we remove props that have been hidden without it being a breaking change?

Comment (nicolethoen) [reply packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:41]: I think that _should_ be okay - but I will defer to @tlabaj here.

**Review by nicolethoen — APPROVED** [review]
As long as it's determined that this is not a breaking change, then it looks good :)

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx (hunk 1 above)
Comment (tlabaj) [reply packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:41]: the hidden props are internal, so consumer should no be using them.  I think it is ok to remove.

**Pull request merged**